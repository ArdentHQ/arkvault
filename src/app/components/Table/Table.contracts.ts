import React from "react";
import { Column, TableState } from "react-table";

export interface TableProperties<RowDataType extends Record<never, unknown>> {
	children?: React.ReactNode | ((data: RowDataType, index: number) => void);
	className?: string;
	data: RowDataType[];
	columns: Column<RowDataType>[];
	hideHeader?: boolean;
	initialState?: Partial<TableState<RowDataType>>;
	rowsPerPage?: number;
	currentPage?: number;
}

export interface TableColumn {
	cellWidth?: string;
	sortDescFirst?: boolean;
	minimumWidth?: boolean;
	disableSortBy?: boolean;
	className?: string;
	headerClassName?: string;
}
