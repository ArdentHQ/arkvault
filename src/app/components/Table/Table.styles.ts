export const defaultTableStyle = `
	table {
		margin: 0;
		padding: 0;
		width: 100%;
		border: none;

		thead {
			th:not(:first-of-type) {
				&:after {
					content: "";
					height: 50%;
					position: absolute;
					left: 0;
					top: 25%;
					border-left-width: 1px;
					border-color: inherit;
				}

				&.no-border:after {
					border-color: transparent;
				}
			}
		}
	}
`;
