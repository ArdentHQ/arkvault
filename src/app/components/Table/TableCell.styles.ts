const baseStyle = "flex items-center px-3 my-1 transition-colors duration-100";

const getHeight = (isCompact: boolean) => (isCompact ? "min-h-11" : "min-h-17.5");

const getVariant = (variant: "start" | "middle" | "end", size?: "sm" | "base"): string | null => {
	if (variant === "start") {
		const variants = {
			base: "pl-6 rounded-l",
			sm: "pl-2 -ml-2 rounded-l",
		};

		return variants[size as keyof typeof variants] || variants.base;
	}

	if (variant === "end") {
		const variants = {
			base: "pr-6 rounded-r",
			sm: "pr-2 -mr-2 rounded-r",
		};

		return variants[size as keyof typeof variants] || variants.base;
	}

	return "";
};

export const getStyles = ({ variant, isCompact, size }: any) =>
	`${baseStyle} ${getHeight(isCompact)} ${getVariant(variant, size)}`;
