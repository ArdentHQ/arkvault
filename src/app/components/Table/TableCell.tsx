import React from "react";
import { styled } from "twin.macro";

import { getStyles } from "./TableCell.styles";

type TableCellProperties = {
	variant?: "start" | "middle" | "end";
	size?: "sm" | "base";
	className?: string;
	innerClassName?: string;
	isCompact?: boolean;
	children: React.ReactNode;
} & Omit<React.HTMLProps<any>, "size">;

const TableCellInnerWrapper = styled.div<TableCellProperties>(getStyles);

export const TableCell = ({
	variant = "middle",
	size,
	className,
	innerClassName,
	isCompact,
	children,
	...properties
}: TableCellProperties) => (
	<td className={className} {...properties}>
		<TableCellInnerWrapper variant={variant} size={size} className={innerClassName} isCompact={isCompact}>
			{children}
		</TableCellInnerWrapper>
	</td>
);
