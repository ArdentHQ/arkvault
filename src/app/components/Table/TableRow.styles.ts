import React from "react";

type TableRowFunction = (event: React.MouseEvent<HTMLTableRowElement, MouseEvent>) => void;

const baseStyle = `transition-colors duration-100`;

const getCursorStyles = (onClick?: TableRowFunction) => onClick && `cursor-pointer`;

const getBorderStyles = (border?: boolean, dotted?: boolean) => {
	if (!border) {
		return;
	}

	return [
		`border-b last:border-b-0 border-theme-secondary-300 dark:border-theme-secondary-800`,
		dotted ? `border-dotted` : `border-dashed`,
	].join(" ");
};

const getHoverStyles = (isSelected?: boolean): string =>
	` table-row ${isSelected ? "table-row-selected" : "table-row-unselected"}`;

export interface TableRowStyleProperties {
	border?: boolean;
	dotted?: boolean;
	onClick?: TableRowFunction;
	isSelected?: boolean;
}

export const getStyles = ({ onClick, border, dotted, isSelected }: TableRowStyleProperties) => {
	const styles: Array<string | undefined> = [baseStyle, getBorderStyles(border, dotted), getCursorStyles(onClick)];

	if (onClick) {
		styles.push(getHoverStyles(isSelected));
	}

	return styles;
};
