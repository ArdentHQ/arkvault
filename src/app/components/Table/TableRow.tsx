import cn from "classnames";
import React from "react";
import { styled } from "twin.macro";

import { getStyles, TableRowStyleProperties } from "./TableRow.styles";

type TableRowProperties = {
	children: React.ReactNode;
} & TableRowStyleProperties &
	React.HTMLProps<any>;

const TableRowStyled = styled.tr<TableRowProperties>(getStyles);

export const TableRow: React.FC<TableRowProperties> = ({
	border = true,
	dotted,
	children,
	isSelected,
	onClick,
	...properties
}) => (
	<TableRowStyled
		data-testid={properties["data-testid"] || "TableRow"}
		className={cn({ group: !!onClick })}
		border={border}
		dotted={dotted}
		isSelected={isSelected}
		onClick={onClick}
	>
		{children}
	</TableRowStyled>
);
