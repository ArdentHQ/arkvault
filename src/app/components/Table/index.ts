export * from "./Table";
export * from "./Table.contracts";
export * from "./TableCell";
export * from "./TableRow";
