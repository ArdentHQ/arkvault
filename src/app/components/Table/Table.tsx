import { chunk } from "@ardenthq/sdk-helpers";
import cn from "classnames";
import React, { useMemo } from "react";
import { HeaderGroup, useSortBy, useTable } from "react-table";
import { styled } from "twin.macro";

import { TableProperties } from "./Table.contracts";
import { defaultTableStyle } from "./Table.styles";
import { Icon } from "@/app/components/Icon";

const TableWrapper = styled.div`
	${defaultTableStyle}
`;

export const Table = <RowDataType extends Record<never, unknown>>({
	children,
	data,
	columns,
	hideHeader = false,
	className,
	initialState,
	rowsPerPage,
	currentPage = 1,
	...properties
}: TableProperties<RowDataType>) => {
	const tableData = useMemo(() => data, [data]);
	const tableColumns = useMemo(() => columns, [columns]);

	const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } = useTable<RowDataType>(
		{
			autoResetSortBy: false,
			columns: tableColumns,
			data: tableData,
			disableSortRemove: true,
			initialState,
		},
		useSortBy,
	);

	const rowsList = useMemo(() => {
		if (!rowsPerPage || rows.length === 0) {
			return rows;
		}

		return chunk(rows, rowsPerPage)[currentPage - 1] || [];
	}, [currentPage, rows, rowsPerPage]);

	const renderChildNode = (rowData: RowDataType, index: number) => {
		if (typeof children === "function") {
			return children(rowData, index);
		}
		return <tr />;
	};

	const renderHeaderGroup = (headerGroup: HeaderGroup<RowDataType>) => (
		<tr
			className="border-b border-theme-secondary-300 dark:border-theme-secondary-800"
			{...headerGroup.getHeaderGroupProps()}
		>
			{headerGroup.headers.map(renderColumn)}
		</tr>
	);

	const renderColumn = (column: HeaderGroup<RowDataType>, thIndex: number) => {
		const thElementClassName = cn(
			"group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold",
			column.headerClassName,
			{ "w-1": column.minimumWidth },
			{
				[`${column.cellWidth} min-${column.cellWidth}`]: !column.minimumWidth && column.cellWidth,
			},
		);
		const rootDivClassName = cn("flex flex-inline align-top", column.className, {
			"flex-row-reverse": column.className?.includes("justify-end") && !column.disableSortBy,
		});
		const iconDivClassName = cn(
			"flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity",
			{ "opacity-0 group-hover:opacity-100": !column.isSorted },
			{
				"ml-2": !column.className?.includes("justify-end"),
				"ml-auto mr-2": column.className?.includes("justify-end"),
			},
		);
		const iconSortClassName: string = cn("transition-transform", {
			"rotate-180": (column.isSorted && !column.isSortedDesc) || (!column.isSorted && !column.sortDescFirst),
		});

		return (
			<th
				className={thElementClassName}
				data-testid={`table__th--${thIndex}`}
				{...column.getHeaderProps(column.getSortByToggleProps())}
			>
				<div className={rootDivClassName}>
					<div className={cn({ "whitespace-nowrap": column.noWrap })}>{column.render("Header")}</div>
					{!column.hideSortArrow && column.canSort && (
						<div className={iconDivClassName}>
							<Icon role="img" name="ChevronDownSmall" className={iconSortClassName} size="sm" />
						</div>
					)}
				</div>
			</th>
		);
	};

	const renderHeader = <thead>{headerGroups.map(renderHeaderGroup)}</thead>;

	return (
		<TableWrapper {...properties} {...getTableProps({ className })} className={cn({ "-mt-3": !hideHeader })}>
			<table cellPadding={0} className="table-auto">
				{!hideHeader && renderHeader}

				<tbody {...getTableBodyProps()}>
					{rowsList.map((row) => {
						prepareRow(row);
						return { ...renderChildNode(row.original, row.index), ...row.getRowProps() };
					})}
				</tbody>
			</table>
		</TableWrapper>
	);
};
