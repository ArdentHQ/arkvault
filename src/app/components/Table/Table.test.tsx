import userEvent from "@testing-library/user-event";
import React from "react";
import { Column } from "react-table";

import { Table } from "./Table";
import { render, screen, within } from "@/utils/testing-library";

const data = [
	{
		col1: "column 1",
		col2: "column 2",
		col3: "column 3",
	},
	{
		col1: "column 1",
		col2: "column 2",
		col3: "column 3",
	},
];

const columns: Column<(typeof data)[number]>[] = [
	{
		Header: "Header 1",
		accessor: "col1",
		className: "flex-row-reverse",
		minimumWidth: true,
	},
	{
		Header: "Header 2",
		accessor: "col2",
		headerClassName: "no-border",
	},
	{
		Header: "Header 3",
		accessor: "col3",
		className: "justify-end",
	},
];

describe("Table", () => {
	it("should render", () => {
		const { container } = render(
			<Table columns={columns} data={data}>
				{() => (
					<tr>
						<td>1</td>
						<td>2</td>
					</tr>
				)}
			</Table>,
		);

		expect(container).toMatchSnapshot();
	});

	it("should render empty rows if template is not provided", async () => {
		const { container } = render(<Table columns={columns} data={[data[0]]} />);

		await expect(within(screen.getAllByRole("rowgroup")[1]).findAllByRole("row")).resolves.toHaveLength(1);
		await expect(within(screen.getAllByRole("rowgroup")[1]).findByTestId("TableRow")).rejects.toThrow(
			/Unable to find an element by/,
		);

		expect(container).toMatchSnapshot();
	});

	it("should change sort order on th click", async () => {
		render(<Table columns={columns} data={data} />);

		const th = screen.getAllByRole("columnheader")[0];

		await userEvent.click(th);

		// eslint-disable-next-line testing-library/no-node-access
		expect(th.querySelector("svg#chevron-down-small")).toBeInTheDocument();

		expect(within(th).getByRole("img")).toHaveClass("rotate-180");

		await userEvent.click(th);

		expect(within(th).getByRole("img")).not.toHaveClass("rotate-180");
	});

	it("should hide header", () => {
		render(<Table hideHeader columns={columns} data={data} />);

		expect(screen.queryAllByRole("columnheader")).toHaveLength(0);
	});

	it("should render with width class applied to column header", () => {
		render(
			<Table
				columns={[
					{
						Header: "Header 1",
						cellWidth: "width",
					},
				]}
				data={data}
			/>,
		);

		expect(screen.getByTestId("table__th--0")).toHaveClass("width");
	});

	it("should split rows", async () => {
		const { container } = render(<Table columns={columns} data={data} rowsPerPage={1} />);

		await expect(within(screen.getAllByRole("rowgroup")[1]).findAllByRole("row")).resolves.toHaveLength(1);

		expect(container).toMatchSnapshot();
	});

	it("should render empty table if pagination returns undefined rows", () => {
		const { container } = render(<Table columns={columns} data={data} rowsPerPage={-1} />);

		expect(container).toMatchSnapshot();
	});
});
