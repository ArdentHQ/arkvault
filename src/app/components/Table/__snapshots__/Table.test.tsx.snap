// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table > should render 1`] = `
<div>
  <div
    class="-mt-3 css-1m6rt2i"
    role="table"
  >
    <table
      cellpadding="0"
      class="table-auto"
    >
      <thead>
        <tr
          class="border-b border-theme-secondary-300 dark:border-theme-secondary-800"
          role="row"
        >
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold w-1"
            colspan="1"
            data-testid="table__th--0"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top flex-row-reverse"
            >
              <div
                class=""
              >
                Header 1
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold no-border"
            colspan="1"
            data-testid="table__th--1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top"
            >
              <div
                class=""
              >
                Header 2
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold"
            colspan="1"
            data-testid="table__th--2"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top justify-end flex-row-reverse"
            >
              <div
                class=""
              >
                Header 3
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-auto mr-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr>
          <td>
            1
          </td>
          <td>
            2
          </td>
        </tr>
        <tr>
          <td>
            1
          </td>
          <td>
            2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Table > should render empty rows if template is not provided 1`] = `
<div>
  <div
    class="-mt-3 css-1m6rt2i"
    role="table"
  >
    <table
      cellpadding="0"
      class="table-auto"
    >
      <thead>
        <tr
          class="border-b border-theme-secondary-300 dark:border-theme-secondary-800"
          role="row"
        >
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold w-1"
            colspan="1"
            data-testid="table__th--0"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top flex-row-reverse"
            >
              <div
                class=""
              >
                Header 1
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold no-border"
            colspan="1"
            data-testid="table__th--1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top"
            >
              <div
                class=""
              >
                Header 2
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold"
            colspan="1"
            data-testid="table__th--2"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top justify-end flex-row-reverse"
            >
              <div
                class=""
              >
                Header 3
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-auto mr-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr />
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Table > should render empty table if pagination returns undefined rows 1`] = `
<div>
  <div
    class="-mt-3 css-1m6rt2i"
    role="table"
  >
    <table
      cellpadding="0"
      class="table-auto"
    >
      <thead>
        <tr
          class="border-b border-theme-secondary-300 dark:border-theme-secondary-800"
          role="row"
        >
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold w-1"
            colspan="1"
            data-testid="table__th--0"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top flex-row-reverse"
            >
              <div
                class=""
              >
                Header 1
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold no-border"
            colspan="1"
            data-testid="table__th--1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top"
            >
              <div
                class=""
              >
                Header 2
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold"
            colspan="1"
            data-testid="table__th--2"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top justify-end flex-row-reverse"
            >
              <div
                class=""
              >
                Header 3
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-auto mr-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      />
    </table>
  </div>
</div>
`;

exports[`Table > should split rows 1`] = `
<div>
  <div
    class="-mt-3 css-1m6rt2i"
    role="table"
  >
    <table
      cellpadding="0"
      class="table-auto"
    >
      <thead>
        <tr
          class="border-b border-theme-secondary-300 dark:border-theme-secondary-800"
          role="row"
        >
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold w-1"
            colspan="1"
            data-testid="table__th--0"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top flex-row-reverse"
            >
              <div
                class=""
              >
                Header 1
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold no-border"
            colspan="1"
            data-testid="table__th--1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top"
            >
              <div
                class=""
              >
                Header 2
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
          <th
            class="group relative text-sm text-left select-none text-theme-secondary-500 border-theme-secondary-300 dark:text-theme-secondary-700 dark:border-theme-secondary-800 m-0 p-3 first:pl-0 last:pr-0 font-semibold"
            colspan="1"
            data-testid="table__th--2"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="flex flex-inline align-top justify-end flex-row-reverse"
            >
              <div
                class=""
              >
                Header 3
              </div>
              <div
                class="flex items-center text-theme-secondary-500 dark:text-theme-secondary-700 transition-opacity opacity-0 group-hover:opacity-100 ml-auto mr-2"
              >
                <div
                  class="transition-transform rotate-180 css-15txs7d"
                  height="10"
                  role="img"
                  width="10"
                >
                  <svg
                    id="chevron-down-small"
                    viewBox="0 0 10 10"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <path
                      d="M9 3.1L5.2 6.9c-.1.1-.3.1-.4 0L1 3.1"
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr />
      </tbody>
    </table>
  </div>
</div>
`;
